plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -817,7 +790,7 @@ public PreparedStatement getPreparedStatement(Connection connection, String sql)
         return connection.prepareStatement(sql);
     }
 
-    protected ResultSet getTables(Connection connection, Optional<String> schemaName, Optional<String> tableName)
+    public ResultSet getTables(Connection connection, Optional<String> schemaName, Optional<String> tableName)
```

Comment:
rename `schemaName` and `tableName` to `reamoteSchemaName` and `remoteTableName`.
Add comment that this method is called by by `IdentifierMapping` and it can not call back to it, as this would result in a loop.

Btw. Can we get rid of this cyclic dependency somehow.

Reply:
I was thinking about extract 3rd component which could be use by jdbc client and identifier mapping. However since this method is overridden by several vendor specific jdbc client it is not that simple. Identifier mapping introduction is quite invasive. So I would leave extracting such component for later...